accepted/2023/wasm-browser-threads.md

```diff
@@ -236,6 +394,10 @@
 - `JSImport` used for logging: `globalThis.console.debug`, `globalThis.console.error`, `globalThis.console.info`, `globalThis.console.warn`, `Blazor._internal.dotNetCriticalError`
     - probably could be any JS context
 
+# WebPack, Rollup friendly
+- it's not clear how to make this single-file
+- because web workers need to start separate script(s)
```

Comment:
single file that behaves differently depending on whether the current context is `window` or not?

Reply:
You can start a WebWorker with a Blob.

Reply:
> You can start a WebWorker with a Blob.

But that's `eval` not CSP friendly. Typically, the people who want single-file also want CSP, because they are doing browser plugin.

Reply:
> single file that behaves differently depending on whether the current context is `window` or not?

For starting the sidecar, we could re-use `dotnet.js` loader. But if both of those scripts were bundled into user code "my-react-application.js" we don't have way how to `new Worker('./dotnet.js')` anymore.
Also, emscripten uses `dotnet.native.worker.js` script and at the moment we don't have nice way how to customize what is inside of it.

I will update the section with these details.